multilog/badger/qry.go

```diff
@@ -133,7 +154,11 @@ func (qry *query) Next(ctx context.Context) (interface{}, error) {
 		}
 	}
 
-	defer func() { qry.nextSeq++ }()
+	if qry.reverse {
+		defer func() { qry.nextSeq-- }()
+	} else {
+		defer func() { qry.nextSeq++ }()
```

Comment:
This looks like an abuse of the defer function :|  maybe add it inline at the end of the function?